cf_xarray/tests/test_accessor.py

```diff
@@ -1237,6 +1237,67 @@ def test_cmip6_attrs():
     assert da.cf.axes["Y"] == ["nlat"]
 
 
+def test_custom_criteria():
+    my_custom_criteria = {
+        "ssh": {
+            "standard_name": "sea_surface_elev*|sea_surface_height",
+            "name": "sea_surface_elevation$",  # variable name
+        },
+        "salt": {
+            "standard_name": "salinity",
+            "name": "sal*",
+        },
+        "wind_speed": {
+            "standard_name": "wind_speed$",
+        },
+    }
+    cf_xarray.accessor.set_options(my_custom_criteria)
```

Comment:
`my_custom_criteria` is set for all the pytest functions called after this, right? Not sure if we want this... 

Also, maybe add a test for `my_custom_criteria` list and/or tuple and/or set as well?

Reply:
I don't see how `my_custom_criteria` could be anything besides a dictionary and still have the necessary structure containing the custom name, attribute to search, and patterns to match with. Can we just limit it to being a dict?

Reply:
I think the way is set up right now, `my_custom_criteria` could be a dictionary or an iterable of dictionaries.
For example, you could pass multiple dictionaries at once. So, using `my_custom_criteria` defined in `test_accessor`, this should work too: `cf_xarray.accessor.set_options([my_custom_criteria])`

I guess limit to `dict` is OK, but at that point I think I'm missing why we are using `ChainMap` under the hood and I guess `OPTIONS["custom_criteria"]` should be just a dict as well?

Reply:
I think we want to support a iterable of dictionaries too.

Reply:
Oh I see, I was trying to imagine how to have a list of the names and patterns, etc, not a super list of dictionaries. I am new to ChainMap and have just been trying to follow what is already in the code for that.

I can add a test for a list and tuple of dictionaries.

Reply:
I just added two tests, one for a list and one for a tuple, but the tuple one doesn't work. If you can see an obvious problem please let me know, otherwise I'll try to figure it out in a few hours when I'm back to this work.

Reply:
I think is this: https://github.com/xarray-contrib/cf-xarray/pull/234#discussion_r656965548

Reply:
I.e., you can't do list + tuple (but you can use list.extend(tuple)).

We decided to go with this implementation though: https://github.com/xarray-contrib/cf-xarray/pull/234#discussion_r657052234

Reply:
Ah, thanks, I see. Ok I will commit @dcherian's implementation and see if the tests run.